x-pack/plugins/session_view/common/utils/expand_dotted_object.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { merge } from '@kbn/std';
+
+const expandDottedField = (dottedFieldName: string, val: unknown): object => {
+  const parts = dottedFieldName.split('.');
+  if (parts.length === 1) {
+    return { [parts[0]]: val };
+  } else {
+    return { [parts[0]]: expandDottedField(parts.slice(1).join('.'), val) };
```

Comment:
Nit: I don't think we'll ever have a key that's nested enough for the performance hit to matter 🤷🏾‍♂️ , but it might be worth creating a separate helper that you can call recursively to handle arrays instead of recreating a flattened key `.join('.')` to only break it apart again on the next recursive call `.split('.')`. You could also update this function to handle that as well with a third param specifically for the split field array, but just thinking the helper might be cleaner.